.github/workflows/run-tests.yaml

```diff
@@ -0,0 +1,106 @@
+name: Run Tests
+on:
+  push:
+  workflow_dispatch:
+    inputs:
+      source:
+        description: 'Source of tests'
+        required: true
+        type: string
+        options:
+          - 'specs'
+          - 'sdk'
+        default: 'sdk'
+      specs_url:
+        description: 'Run tests based on specs'
+        required: false
+        type: string
+        default: ''
+      sdk_version:
+        description: 'Run tests based on SDK'
+        required: false
+        type: string
+        default: 'LATEST'
+      sdk_namespace:
+        description: 'SDK to test'
+        required: true
+        type: string
+        options:
+          - rapid
+        default: 'rapid'
+      endpoint_prefix:
+        description: 'Endpoint to prepend specs paths with'
+        required: true
+        type: string
+      jdk:
+        description: 'JDK version to use'
+        required: true
+        type: string
+        options:
+          - '17'
+          - '21'
+        default: '17'
+  workflow_call:
+    inputs:
+      source:
+        description: 'Source of tests'
+        required: true
+        type: string
+        default: 'sdk'
+      specs_url:
+        description: 'Run tests based on specs'
+        required: false
+        type: string
+        default: ''
+      sdk_version:
+        description: 'Run tests based on SDK'
+        required: false
+        type: string
+        default: 'LATEST'
+      sdk_namespace:
+        description: 'SDK to test'
+        required: true
+        type: string
+        default: 'rapid'
+      jdk:
+        description: 'JDK version to use'
+        required: true
+        type: string
+        default: '17'
+      endpoint_prefix:
+        description: 'Endpoint to prepend specs paths with'
+        required: true
+        type: string
+
+jobs:
+  inputs-validation:
+    runs-on: ubuntu-latest
+    steps:
+      - shell: python -u {0}
+        run: |
+          if any([
+            'specs' in '${{ github.event.inputs.source }}' and not('${{ github.event.inputs.specs_url }}'),
+            'sdk' in '${{ github.event.inputs.source }}' and not('${{ github.event.inputs.sdk_version }}')
+          ]):
+            print('::error::Invalid inputs')
+            exit(1)
+  generate-test-sdk:
+    if: ${{ github.event.inputs.source == 'specs' }}
+    needs: [ inputs-validation ]
+    uses: ./.github/workflows/generator-test-sdk.yaml
+    with:
+      version: ${{ github.event.inputs.sdk_version }}
+      namespace: ${{ github.event.inputs.sdk_namespace }}
+      endpoint_prefix: ${{ github.event.inputs.endpoint_prefix }}
+    secrets: inherit
+  run-rapid-examples:
+    if: always() && !contains(needs.*.result, 'failure') && !contains(needs.*.result, 'cancelled')
```

Comment:
Why the `always` check here? It doesn't make sense to me...

Reply:
The always keyword here will override the implicit condition that dependency actions will execute.